Project/Src/Gui/TextEditorControlBase.cs

```diff
@@ -208,9 +208,11 @@ private void GenerateDefaultActions()
             editactions[Keys.Up] = new CaretUp();
             editactions[Keys.Up | Keys.Shift] = new ShiftCaretUp();
             editactions[Keys.Up | Keys.Control] = new ScrollLineUp();
+            editactions[Keys.Up | Keys.Alt | Keys.Shift] = new MoveLineUp();
```

Comment:
Why not ALT+Up/Down? The same as what most of other editors define.

Reply:
Happy you ask a question because I planned to discuss about that even if I didn't had the time to add a comment.
I also prefer ALT+Up/Down (simpler and more frequent) but it's already used **in read only mode** to navigate between occurrences of selected text.

So it is possible to use it to do something else in edit mode but it requires some more code in the editor (not this control!) otherwise the event is 'swallowed'. I spent quite some time trying to understand that but I now have it working.

So it's doable when the submodule change will be included in GE.

Just that the same shortcut will be used for 2 different things in readonly and edit modes. I don't know if it is unacceptable...

Reply:
> but it's already used **in read only mode** to navigate between occurrences of selected text.

Can we re-assign that?

Reply:
Yes even if I didn't want to break someone workflow. But I should admit that it should not be very used. I discovered it digging the code...
Other thing is that we should find good new shortcuts...

Reply:
> but it's already used **in read only mode** to navigate between occurrences of selected text.

Looking at the code again, it's the default hotkey to navigate between diff *in diff mode**.
I don't think we should change it (and anyway as it is configurable we will never sure it won't be change back) because it is consistent with some other places where we use the same hotkey and also navigating between occurrences.

![image](https://github.com/gitextensions/ICSharpCode.TextEditor/assets/460196/29a504b4-927c-42df-ae34-9b62e2fdbd15)

But, as one is available in diff mode and the other in edit mode, I don't think there is really a problem.

We will just have to wrap the PerformClick by doing that: https://github.com/gitextensions/gitextensions/commit/d92334aa70943866c242c7caf3a6e4ca326f6e3b

So, I'm ok to change to ALT+Up/Down (that I prefer also because I'm used to...)

Reply:
The diff viewer is a readonly control, right? I think it's totally acceptable to use the same shortcut for a completely different control in a different context. I also think that anyone who uses IDEs of any kind on a regular basis would have any issues with this.

@gerhardol @mstv thoughts?

Reply:
I have made the changes while getting others point of views...

Reply:
I agree this should be changed.

But the most important usage as I see it is in rebase dialog, that should be replaced with a form.